### PR TITLE
Support for complex matrices

### LINKED ISSUE #16: Complex input not supported
Hi all,

It seems that complex input is not supported.
```
Julia> v1 = [1.0+1.0im,2.,3.]
Julia> v2 = [1.0+1.0im,5.0+1.1im,6.]
Julia> A = Toeplitz(v1, v2)
ERROR: LoadError: MethodError: `convert` has no method matching convert(::Type{ToeplitzMatrices.Toeplitz{T<:Number}}, ::Array{Complex{Float64},1}, ::Array{Complex{Float64},1}, ::Array{Complex{Float64},1}, ::Array{Complex{Float64},1}, ::Base.DFT.FFTW.cFFTWPlan{Complex{Float64},-1,true,1})
This may have arisen from a call to the constructor ToeplitzMatrices.Toeplitz{T<:Number}(...),
since type constructors fall back to convert methods.
Closest candidates are:
  ToeplitzMatrices.Toeplitz{T<:Number}(::Array{T<:Number,1}, ::Array{T<:Number,1}, ::Array{Complex{T<:Number},1}, ::Array{Complex{T<:Number},1}, ::Base.DFT.Plan{Complex{T<:Number}})
  ToeplitzMatrices.Toeplitz{T<:Number}(::Array{T<:Number,1}, ::Array{T<:Number,1})
  ToeplitzMatrices.Toeplitz{T1<:Number,T2<:Number}(::Array{T1<:Number,1}, ::Array{T2<:Number,1})
  ...
 in call at /home/kuan/Documents/git/ToeplitzMatrices.jl/src/ToeplitzMatrices.jl:120
 in include at ./boot.jl:261
 in include_from_node1 at ./loading.jl:320
while loading /home/kuan/Documents/git/test/test4.jl, in expression starting on line 32
```
The offending line is Line 120 in ToeplitzMatrices.jl, which calls the ctor but mismatches the type. 

I wonder the best way to fix this. Any ideas please?

### PR DESCRIPTION
**Pull request by andreasnoack**
Enable travis testing with BigFloat

Fixes #16

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -77,40 +80,62 @@ Hl = Hankel(0.9.^(nl-1:-1:0), 0.4.^(0:nl-1))
...
+print("Complex rectangular: ")
+Hs = Hankel(complex(0.9.^(ns-1:-1:0)), complex(0.4.^(0:nl-1)))
+Hl = Hankel(complex(0.9.^(nl-1:-1:0)), complex(0.4.^(0:ns-1)))
+@test_approx_eq Hs * xl[:,1] full(Hs) * xl[:,1]
+@test_approx_eq Hs * xl full(Hs) * xl
+@test_approx_eq Hl * xs full(Hl) * xs
+println("OK!")
+
 
 if isdir(Pkg.dir("FastTransforms"))
```
Comment (tkelman) [inline test/runtests.jl]: should probably remove this condition if it's in `test/REQUIRE` now

**Comment by dlfivefifty** [comment]
Since FastTransforms depends on ToeplitzMatrices, maybe it makes more sense to move the tests there?

**Comment by tkelman** [comment]
depends which package's changes you think are more likely to break them - could put them both places if you felt so inclined

**Comment by dlfivefifty** [comment]
The tests make more sense here, since it's just using the BigFloat FFT implemented in FastTransforms.

**Pull request merged**